### PR TITLE
Link payment flows to form on landing page

### PR DESCRIPTION
**Pull request by solarissmoke**
This PR links up the payment flows developed earlier to the form on the landing page, so that you can select an amount from that page, and continue through to payment (either single or monthly). It also provides initial support for currency switching.

### CODE REVIEW

File: donate/templates/payment/personal_details.html [hunk 1]
```diff
@@ -11,15 +11,15 @@ <h1>{% trans "Your details" %}</h1>
...
+            <p id="donate__heading--single">{% trans "You are donating" %} {{ currency_info.symbol }}&nbsp;<span id="donate__amount__display">{{ form.amount.value|floatformat:"-2" }}</span></p>
         </div>
 
         <div class="donate__amount" id="donate__amount">
             <a id="donate__change-amount" href="#donate__amount-edit">Change amount</a>
         </div>
 
         <div class="donate__bar-edit" id="donate__amount-edit" hidden>
-            <span class="form__input__prefix">$</span> <input type="number" value="{{ form.amount.value }}" step="1" min="1">
+            <span class="form__input__prefix">{{ currency_info.symbol }}</span> <input type="number" value="{{ form.amount.value }}" step="1" min="1">
```
Comment (TheoChevalier) [inline donate/templates/payment/personal_details.html]: A lot of locales will want the currency symbol to be displayed after the value, would it be possible to add a conditional for that? Maybe storing that preference in settings?

To give you an idea: https://transvision.mozfr.org/string/?entity=mozilla_org/mozorg/home/index-quantum.lang:3fe118b0ed61665a878f4a81ef2d3357&repo=mozilla_org

Comment (TheoChevalier) [reply donate/templates/payment/personal_details.html]: I expect there are other places where having this preference will be handy

**Comment by solarissmoke** [comment]
@nicklee would you be able to review the JS changes made here?

File: donate/templates/payment/personal_details.html (hunk 1 above)
Comment (solarissmoke) [reply donate/templates/payment/personal_details.html]: OK, thanks. I am going to make a separate task to work on this, just to keep this change contained.

**Comment by jafacakes2011** [comment]
LGTM

**Pull request merged**